compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -2133,8 +2133,34 @@ trait Applications extends Compatibility {
     }
   }
 
-  def isApplicableExtensionMethod(ref: TermRef, receiver: Type)(using Context) =
-    ref.symbol.is(ExtensionMethod)
-    && !receiver.isBottomType
-    && isApplicableMethodRef(ref, receiver :: Nil, WildcardType)
+  private def tryApplyingReceiverToTruncatedExtMethod(methodSym: TermSymbol, receiver: Tree)(using Context): scala.util.Try[Tree] =
```

Comment:
Usage of `TermSymbol` cannot handle prefix types, it's better to use `TermRef`.

The name of the method is long, but not informative. I'd suggest make it shorter and add documentation to make its semantics clear.